kolibri/plugins/learn/assets/src/state/actions.js

```diff
@@ -447,10 +447,17 @@ function showSearch(store, channelId, searchTerm) {
 }
 
 function showExamList(store, channelId) {
-  store.dispatch('CORE_SET_PAGE_LOADING', true);
+  const userIsLoggedIn = coreGetters.isUserLoggedIn(store.state);
   store.dispatch('SET_PAGE_NAME', PageNames.EXAM_LIST);
+  store.dispatch('CORE_SET_PAGE_LOADING', true);
 
-  coreActions.setChannelInfo(store, channelId).then(
+  // if user is not logged in, this action is a noop
+  if (!userIsLoggedIn) {
+    store.dispatch('CORE_SET_PAGE_LOADING', false);
```

Comment:
Can we do the redirect inside the router, before we even hit this action? This would obviate a lot of this code actually, although it's debatable whether it is good to just redirect a user without explanation.

I could dispatch an error message to show in the red box, but it seems like the "did you forget to login" message should be enough explanation.